app/models/setting/email.rb

```diff
@@ -0,0 +1,72 @@
+require 'facter'
+
+class Setting::Email < Setting
+  WHITELIST = %w(send_welcome_email email_reply_address email_subject_prefix)
+
+  def self.load_defaults
+    # Check the table exists
+    return unless super
+    domain = Facter.value(:domain) || SETTINGS[:domain]
+    email_reply_address = "Foreman-noreply@#{domain}"
+
+    self.transaction do
+      [
+        self.set('email_reply_address', N_("Email reply address for emails that Foreman is sending"), email_reply_address, N_('Email reply address')),
+        self.set('email_subject_prefix', N_("Prefix to add to all outgoing email"), '[foreman]', N_('Email subject prefix')),
+        self.set('send_welcome_email', N_("Send a welcome email including username and URL to new users"), false, N_('Send welcome email')),
+        self.set('delivery_method', N_("Method used to deliver e-mail"), 'sendmail', N_('Delivery method'), nil, { :collection => Proc.new {{'sendmail' => :sendmail, 'smtp' => :smtp}}}),
```

Comment:
a) "e-mail" has no hyphen in Foreman
b) "smtp" should be "SMTP", it's an acronym